apps/prairielearn/src/migrations/20230531184500_assessment_access_policies.sql

```diff
@@ -0,0 +1,19 @@
+-- Create assessment_access_policies table
+CREATE TABLE IF NOT EXISTS
+  assessment_access_policies (
+    id BIGSERIAL PRIMARY KEY,
+    assessment_id BIGINT REFERENCES assessments (id) ON UPDATE CASCADE ON DELETE CASCADE NULLS NOT DISTINCT,
+    user_id BIGINT REFERENCES users (user_id) ON UPDATE CASCADE ON DELETE CASCADE NULLS NOT DISTINCT,
+    group_id BIGINT REFERENCES groups (id) ON UPDATE CASCADE ON DELETE CASCADE NULLS NOT DISTINCT,
```

Comment:
`NULLS NOT DISTINCT` doesn't have any meaning for non-unique columns. In fact, it's a parser error in its current form. Is this new code tested at all?

@mwest1066 resurfacing an earlier discussion - do we want the combination of `(assessment_id, user_id)` (same with `group_id`) to be unique? That is, should we allow multiple overrides for one student/group?

Reply:
From the other comment, I think we do _not_ want multiple overrides for a given student/group, so we _do_ want `(assessment_id, user_id)` to be unique (and similarly for `group_id`).